common/azure-common/src/main/java/com/azure/common/annotations/UnexpectedResponseExceptionType.java

```diff
@@ -1,37 +1,51 @@
-/**
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
- */
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 package com.azure.common.annotations;
 
 import com.azure.common.exception.ServiceRequestException;
 
+import java.lang.annotation.Repeatable;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 import static java.lang.annotation.ElementType.METHOD;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 /**
- * The error type that will be thrown or returned when an unexpected status code is returned from an REST API.
+ * The exception type that is thrown or returned when one of the status codes is returned from a REST API. Multiple
+ * annotations can be used, they are processed in a top-down order. When no codes are listed that exception is always
+ * thrown or returned if it is reached during evaluation, this should be treated as a default case. If no default case
+ * is annotated the fall through exception is ServiceRequestException.
  *
  * <p><strong>Example:</strong></p>
  *
  * <pre>
  * {@literal @}UnexpectedResponseExceptionType(MyCustomException.class)
  * {@literal @}POST("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/images/getEntityTypeImageUploadUrl")
  *  void getUploadUrlForEntityType(@Path("resourceGroupName") String resourceGroupName, @Path("hubName") String hubName, @Path("subscriptionId") String subscriptionId, @Body GetImageUploadUrlInputInner parameters);
+
+ * {@literal @}UnexpectedResponseExceptionType(code = {404}, value = NotFoundException.class)
+ * {@literal @}UnexpectedResponseExceptionType(DefaultException.class)
+ * {@literal @}POST("subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/images/getEntityTypeImageUploadUrl")
+ *  void getUploadUrlForEntityType(@Path("resourceGroupName") String resourceGroupName, @Path("hubName") String hubName, @Path("subscriptionId") String subscriptionId, @Body GetImageUploadUrlInputInner parameters);
  * </pre>
  */
 @Retention(RUNTIME)
 @Target(METHOD)
+@Repeatable(UnexpectedResponseExceptionTypes.class)
```

Comment:
This could probably just be the singular type `UnexpectedResponseExceptionType`

Reply:
This is required as the Repeatable annotation needs a class that has a value method that returns an array of the class its annotating, in this case `UnexpectedResponseExceptionType[] value()`.